### PR TITLE
Fix update whitelist data race

### PR DESCRIPTION
**Pull request by aneequesafdar**

### CODE REVIEW

File: consensus/test/tendermint_test.go [hunk 1]
```diff
@@ -875,25 +875,6 @@ func runTest(t *testing.T, test *testCase) {
 		t.Fatal(err)
 	}
 
-	wg = &errgroup.Group{}
-	for i, validator := range validators {
-		validator := validator
-		i := i
-
-		wg.Go(func() error {
-			log.Debug("peers", "i", i,
```
Comment (JekaMas) [inline consensus/test/tendermint_test.go]: Why have you removed this test info?

File: consensus/tendermint/core/core.go [hunk 2]
```diff
@@ -302,18 +305,10 @@ func (c *core) setCore(r *big.Int, h *big.Int, lastProposer common.Address) {
...
-	}
-	prevoteTime := newTimeout(prevote)
-	if ok := c.prevoteTimeout.set(prevoteTime); !ok {
-		c.prevoteTimeout = prevoteTime
-	}
-	precommitTime := newTimeout(precommit)
-	if ok := c.precommitTimeout.set(precommitTime); !ok {
-		c.precommitTimeout = precommitTime
-	}
+	c.proposeTimeout.reset(propose)
```
Comment (JekaMas) [inline consensus/tendermint/core/core.go]: Could you write a test with concurrent reset and get timeout usage?

**Comment by aneequesafdar** [comment]
Yeah i think I am going to wait till Boris fix his dead lock since I was also getting a deadlock.

**Comment by aneequesafdar** [comment]
closing in favour of #300

**Pull request closed without merging**